app/scripts/modules/core/src/deploymentStrategy/strategies/monitored/AdditionalFields.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+
+import { IDeploymentStrategyAdditionalFieldsProps } from 'core/deploymentStrategy/deploymentStrategy.registry';
+import { HelpField } from 'core/help/HelpField';
+import { NgReact } from 'core/reactShims';
+import { IServerGroupCommand } from 'core/serverGroup';
+import Select, { Option } from 'react-select';
+import {
+  DeploymentMonitorReader,
+  IDeploymentMonitorDefinition,
+} from 'core/pipeline/config/stages/monitoreddeploy/DeploymentMonitorReader';
+
+import { set } from 'lodash';
+
+export interface IMonitoredDeployCommand extends IServerGroupCommand {
+  delayBeforeScaleDownSec: string;
+  rollback: {
+    onFailure: boolean;
+  };
+  maxRemainingAsgs: number;
+  scaleDown: boolean;
+  deploySteps: number[] | string;
+  deploymentMonitor: {
+    id: string;
+    parameters: {};
+  };
+}
+
+export interface IMonitoredDeployStrategyAdditionalFieldsProps extends IDeploymentStrategyAdditionalFieldsProps {
+  command: IMonitoredDeployCommand;
+}
+
+export interface IMonitoredDeployStrategyAdditionalFieldsState {
+  deploymentMonitors: IDeploymentMonitorDefinition[];
+}
+
+export class AdditionalFields extends React.Component<
+  IMonitoredDeployStrategyAdditionalFieldsProps,
+  IMonitoredDeployStrategyAdditionalFieldsState
+> {
+  public state: IMonitoredDeployStrategyAdditionalFieldsState = {
+    deploymentMonitors: [],
+  };
+
+  public componentDidMount() {
+    DeploymentMonitorReader.getDeploymentMonitors().then(deploymentMonitors => {
+      this.setState({ deploymentMonitors });
+    });
+  }
+
+  private deployStepsChange = (model: number[] | string) => {
+    this.props.command.deploySteps = model;
+    this.forceUpdate();
+  };
+
+  private scaleDownChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.props.command.scaleDown = e.target.checked;
+    this.forceUpdate();
+  };
+
+  private rollbackOnFailureChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.props.command.rollback.onFailure = e.target.checked;
+    this.forceUpdate();
+  };
+
+  private maxRemainingAsgsChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.props.command.maxRemainingAsgs = parseInt(e.target.value, 10);
+    this.forceUpdate();
+  };
+
+  private handleDeploymentMonitorChange = (option: Option<string>) => {
+    this.props.command.deploymentMonitor.id = option.value;
+    this.forceUpdate();
+  };
+
+  private handleChange = (key: string, value: string) => {
+    set(this.props.command, key, value);
+    this.forceUpdate();
+  };
+
+  public render() {
+    const { NumberList } = NgReact;
+    const { command } = this.props;
+    const rollbackOnFailure = command.rollback && command.rollback.onFailure;
+
+    return (
+      <div className="form-group">
+        {this.state.deploymentMonitors && (
```

Comment:
do you want to learn the newer and totally better way to do forms that is definitely the final way we will ever do forms?

Reply:
yes... someday... :)?